src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/NVidiaEmbeddingClient.java

```diff
@@ -65,8 +85,19 @@ public Uni<List<float[]>> vectorize(
     EmbeddingRequest request =
         new EmbeddingRequest(texts.toArray(textArray), modelName, input_type);
     Uni<EmbeddingResponse> response =
-        embeddingProvider.embed(
-            "Bearer " + (apiKeyOverride.isPresent() ? apiKeyOverride.get() : apiKey), request);
+        embeddingProvider
+            .embed(
+                "Bearer " + (apiKeyOverride.isPresent() ? apiKeyOverride.get() : apiKey), request)
+            .onFailure(
+                throwable -> {
+                  return (throwable.getCause() != null
```

Comment:
Is the `JsonApiException` always wrapped inside something else?

Reply:
Good catch., fixed it.